features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Operations/Operations.jsx

```diff
@@ -0,0 +1,440 @@
+        /* eslint no-underscore-dangle: ["error", { "allow": ["_data"] }] */
+        /* eslint no-param-reassign: ["error", { "props": false }] */
+        if (api._data) delete api._data;
+        if (api.client) delete api.client;
+        if (api.apiType) delete api.apiType;
+        const promisedApi = this.newApi.update(JSON.parse(JSON.stringify(api)));
+        promisedApi
+            .then(() => {
+                Alert.info(intl.formatMessage({
+                    id: 'Apis.Details.Operations.Operations.api.updated.successfully',
+                    defaultMessage: 'API updated successfully!',
+                }));
+            })
+            .catch((error) => {
+                console.error(error);
+                Alert.error(intl.formatMessage({
+                    id: 'Apis.Details.Operations.Operations.something.went.wrong.while.updating.the.api',
+                    defaultMessage: 'Error occurred while updating API',
+                }));
+            });
+    }
+    toggleAssignScopes = () => {
+        // this.setState({ showScopes: !this.state.showScopes });
+        this.setState(prevState => ({ showScopes: !prevState.showScopes }));
+    }
+
+    /**
+     * @inheritdoc
+     */
+    render() {
+        const {
+            operationList, scopes, showScopes, apiPolicies,
+        } = this.state;
+        if (this.state.notFound) {
+            return <ResourceNotFound message={this.props.resourceNotFountMessage} />;
+        }
+        if (!operationList) {
+            return <Progress />;
+        }
+        const { classes } = this.props;
+        return (
+            <div className={classes.root}>
+                <div className={classes.titleWrapper}>
+                    <Typography variant='h4' align='left' className={classes.mainTitle}>
+                        <FormattedMessage
+                            id='Apis.Details.Operations.Operations.operation'
+                            defaultMessage='Operations'
+                        />
+                    </Typography>
+                    <Button size='small' className={classes.button} onClick={this.toggleAssignScopes}>
+                        <ScopesIcon className={classes.buttonIcon} />
+                        <FormattedMessage
+                            id='Apis.Details.Resources.Resources.assign.global.scope.for.api'
+                            defaultMessage='Assign Global Scope for API'
+                        />
+                    </Button>
+                </div>
+                <div className={classes.contentWrapper}>
+                    {(this.state.scopes && showScopes) &&
+                        <React.Fragment>
+                            <div className={classes.addNewWrapper}>
+                                <Typography className={classes.addNewHeader}>
+                                    <FormattedMessage
+                                        id='Apis.Details.Resources.Resources.assign.global.scopes.for.api.title'
```

Comment:
id violating the agreed format.

Reply:
Removed with f95c97a9d7d71310edcbe867005d8d9b3bac696d